### PR TITLE
add support for Tables

### PR DESCRIPTION
**Pull request by bardurdam**
- add `Table`, `TableColumn`, `TableStyle` and `TableStyleName` to
  model.

- make `Model2XlsxConversions` build tables from model.

- add example for defining a table

- tell Travis to use JDK8

### CODE REVIEW

File: src/main/scala/com/norbitltd/spoiwo/model/Table.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    cellRange: CellRange,
+    id: java.lang.Long = null,
+    name: String = null,
+    displayName: String = null,
+    columns: List[TableColumn] = Nil,
+    style: TableStyle = null,
+    enableAutoFilter: java.lang.Boolean = null
+  ): Table =
+    Table(
+      cellRange,
```
Comment (norbert-radyk) [inline src/main/scala/com/norbitltd/spoiwo/model/Table.scala]: Please use named parameters as in i.e. `Sheet`. They do add a bit code in a short run, but are much safer especially with many attributes of the same type (i.e. you will not mix up name and display name by accident).

File: src/main/scala/com/norbitltd/spoiwo/natures/xlsx/Model2XlsxConversions.scala [hunk 2]
```diff
@@ -274,9 +274,64 @@ object Model2XlsxConversions {
     s.repeatingRows.foreach(rr => sheet.setRepeatingRows(convertRowRange(rr)))
     s.repeatingColumns.foreach(rc => sheet.setRepeatingColumns(convertColumnRange(rc)))
     s.password.foreach(ps => sheet.protectSheet(ps))
+    s.tables.foreach(tbl ⇒ convertTable(tbl, sheet))
     sheet
   }
 
+  private[xlsx] def convertTable(modelTable: Table, sheet: XSSFSheet): Unit = {
```
Comment (norbert-radyk) [inline src/main/scala/com/norbitltd/spoiwo/natures/xlsx/Model2XlsxConversions.scala]: Where is the Spec for this conversion?

File: src/main/scala/com/norbitltd/spoiwo/natures/xlsx/Model2XlsxConversions.scala [hunk 3]
```diff
@@ -274,9 +274,64 @@ object Model2XlsxConversions {
...
+    cttable.setRef(reference.formatAsString())
+
+
+    // AutoFilter
+    if (modelTable.enableAutoFilter.contains(true))
+      cttable.addNewAutoFilter()
+
+
+    // Columns
+    val modelColumns = if(modelTable.columns.isEmpty) {
```
Comment (norbert-radyk) [inline src/main/scala/com/norbitltd/spoiwo/natures/xlsx/Model2XlsxConversions.scala]: So you are checking here if the `modelTable.columns` are empty and then fill with the number of columns from start to end. What will happen if `modelTable.columns.size < (eCol - sCol) + 1`

**Comment by bardurdam** [comment]
@norbert-radyk, thanks for the feedback.

I have updated the PR.

I have set a validation wrt. table columns, so if they are specified, but do not equal the amount of needed columns, you will get a runtime exception. Table columns are therefore only generated when not speficied at all.

I've added specs too.

**Pull request merged**